app/questionnaire/navigator.py

```diff
@@ -24,7 +24,7 @@ def evaluate_rule(rule, answer):
     return False
 
 
-def build_path(blocks, block_id, answers, path):
+def build_path(blocks, block_id, metadata, answers, path):
```

Comment:
The navigator has changed quite a bit for the repeating work - it no longer takes answers as method arguments and uses them when initialised. If we're having to evaluate metadata too, it probably makes sense for that to be an init arg too.